### PR TITLE
[DPE-4334] Add a product entry for Spark 3.5.1

### PR DESCRIPTION
**Pull request by theoctober19th**

### CODE REVIEW

File: products.json [hunk 1]
```diff
@@ -1,4 +1,18 @@
 [
+    {
+        "name": "spark",
+        "lp-building-project": "soss",
+        "lp-releasing-project": "spark-releases",
+        "lp-building-repo": "~data-platform/soss/+source/charmed-spark/+git/charmed-spark/",
+        "lp-building-branch-prefix": "lp-3.5.1",
```
Comment (welpaolo) [inline products.json:7]:
```suggestion
        "lp-building-branch-prefix": "lp-3.5",
```

**Review by welpaolo — COMMENTED** [review]
Please fix branch prefix

File: products.json [hunk 2]
```diff
@@ -1,4 +1,18 @@
 [
+    {
+        "name": "spark-3.5.1",
```
Comment (deusebio) [inline products.json:3]: @welpaolo should we call this Spark 3.5 and also change the name of the other one from "spark" to "spark-3.4"?

Does this name matter?

Comment (theoctober19th) [reply products.json]: Renamed to `spark-3.5` and `spark-3.4`.

**Review by welpaolo — APPROVED** [review]

**Review by deusebio — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**